src/Contract/Entity/TimestampableInterface.php

```diff
@@ -8,13 +8,27 @@
 
 interface TimestampableInterface
 {
+    /**
+     * Updates timestamps for createdAt and updatedAt properties.
+     */
     public function updateTimestamps(): void;
 
-    public function getCreatedAt(): DateTimeInterface;
-
-    public function setCreatedAt(DateTimeInterface $createdAt): void;
-
-    public function getUpdatedAt(): DateTimeInterface;
-
-    public function setUpdatedAt(DateTimeInterface $updatedAt): void;
+    /**
+     * Returns a DateTimeInterface object with the current timestamp.
+     */
+    public function getCurrentDateTime(): DateTimeInterface;
+
+    /**
+     * Returns an array of createdAt properties.
+     *
+     * @return string[]
+     */
+    public static function getCreatedAtProperties(): array;
+
+    /**
+     * Returns an array of updatedAt properties.
+     *
+     * @return string[]
+     */
+    public static function getUpdatedAtProperties(): array;
```

Comment:
How could we get rid of static? These method are available on entity object

Reply:
I need it in the event subscriber which is why they are static as I don't have an object there. If you have a better idea though...

Reply:
I'd get object first somehow and work with that.

Reply:
I will think about a better solution. what about an annotation on the properties instead?

Reply:
@TomasVotruba I have now made them non-static.
I thought about using annotations but the issue then is, I also would need to access the data inside the class itself and I didn't want to read annotations inside the entity, so I made them non static.

However, Doctrine has already a method to create an instance of the entity (in the classmetadata) without executing the constructor, so it was easy to create an instance. If not, I could also have used reflection but if Doctrine already has a method, I think it is better to use that one (e.g. caching etc.).